### ISSUE TITLE
JAR Fails to Open

### ISSUE THREAD

**Issue by SeaMac** [agateau/pixelwheels#78]
My MacPro5,1 (OS 10.13.6) won't run the Pixelwheels .jar ... 
Error messages from the Failed Launch are:
02:06:16.933744 -0800	Finder	com.apple.message.domain: com.apple.finder.view_style
com.apple.message.signature: 3
com.apple.message.summarize: YES
SenderMachUUID: 396F2916-7331-3D7C-8019-7EF8E1299CBA

02:06:46.971605 -0800	Finder	OSActivityID: 0x800000000001868b
com.apple.message.domain: com.apple.open.document
com.apple.message.signature: Finder
com.apple.message.summarize: YES
SenderMachUUID: 396F2916-7331-3D7C-8019-7EF8E1299CBA

02:06:47.110204 -0800	Finder	OSActivityID: 0x800000000001868b
com.apple.message.domain: com.apple.coreservices.applaunch
com.apple.message.path_bucket: 6
com.apple.message.volume_is_network: no
com.apple.message.quarantined: no
com.apple.message.volume_is_removable: no
com.apple.message.volume_is_ejectable: no
com.apple.message.volume_is_root: yes
com.apple.message.volume_is_disk_image: no
com.apple.message.bundle_identifier: com.apple.JarLauncher
com.apple.message.bundle_version: 15.0.1
com.apple.message.bundle_short_version: 15.0.1
SenderMachUUID: 396F2916-7331-3D7C-8019-7EF8E1299CBA

02:06:47.132871 -0800	CoreServicesUIAgent	com.apple.message.domain: com.apple.usage.gatekeeper.open-unsafe-executable-type
com.apple.message.signature: com.sun.java-archive
com.apple.message.signature2: LSRiskCategoryUnsafeExecutable
SenderMachUUID: 3FCB9D91-2929-353A-9031-9AE8D7687CB1

02:06:47.195362 -0800	syspolicyd	assessment denied for pixelwheels.jar
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature3: pixelwheels.jar
com.apple.message.signature5: UNKNOWN
com.apple.message.signature: denied:no usable signature
com.apple.message.signature4: 3
SenderMachUUID: 0C920F04-0299-329C-810B-8ED1557532C0

02:06:50.015461 -0800	syspolicyd	added rule for pixelwheels.jar
com.apple.message.domain: com.apple.security.assessment.update
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature3: pixelwheels.jar
com.apple.message.signature5: UNKNOWN
SenderMachUUID: 0C920F04-0299-329C-810B-8ED1557532C0

02:06:50.015754 -0800	CoreServicesUIAgent	added rule for pixelwheels.jar
com.apple.message.domain: com.apple.security.assessment.update
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature3: pixelwheels.jar
com.apple.message.signature5: UNKNOWN
SenderMachUUID: 3FCB9D91-2929-353A-9031-9AE8D7687CB1

02:07:04.085938 -0800	Unknown	com.apple.message.domain: com.apple.assumes.failure
com.apple.message.signature: D1E75C38-62CE-3D77-9ED3-5F6D38EF0676:914800
com.apple.message.signature2: 0x40
com.apple.message.signature3: systemstats
com.apple.message.summarize: YES
SenderMachUUID: D1E75C38-62CE-3D77-9ED3-5F6D38EF0676

02:07:09.323294 -0800	Dock	com.apple.message.domain: com.apple.open.application
com.apple.message.signature: Dock
com.apple.message.summarize: YES
SenderMachUUID: 2FA55534-87F6-3967-8AFF-589C0DA1467E

02:07:09.324823 -0800	Dock	com.apple.message.domain: com.apple.coreservices.applaunch
com.apple.message.path_bucket: 0
com.apple.message.volume_is_network: no
com.apple.message.volume_is_removable: no
com.apple.message.volume_is_ejectable: no
com.apple.message.quarantined: no
com.apple.message.volume_is_root: yes
com.apple.message.volume_is_disk_image: no
com.apple.message.bundle_identifier: com.apple.Console
com.apple.message.bundle_version: 2.0.52
com.apple.message.bundle_short_version: 1.0
SenderMachUUID: 2FA55534-87F6-3967-8AFF-589C0DA1467E

It looks as if the Gatekeeper stopped it: but I right Clicked and chose "Open" from the Finder. I gave it permission, AFAIK

**Comment by agateau** [comment]
Oh, I must confess I haven't tested Pixel Wheels on macOS recently. Will give it a try when I get access to the Macbook Pro I use to test.

**Comment by SeaMac** [comment]
Thanks. I had wanted to post a link to that game, on my site, but decided to test it out on my system first. ("Burger Party" worked OK and got posted that same day - an inbound link!!! https://softwaresanta.com/sssmf/index.php/topic,5717.0.html ) I love mature open source games where the developer builds for multiple systems - keep up the good work. (I don't code so well - BASIC is my preferred interpreter - but I know how to file a decent bug report.)
I'll be your Mac Beta Tester ... in parallel with YOU, of course - if you wish.

**Comment by agateau** [comment]
Sorry for the late reply. Having more macOS testers would definitely be a good thing!

I finally took the time to test on macOS 10.15.2. I was able to start the game from the command line but it crashed with an error message suggesting to start it with `-XstartOnFirstThread`. This option makes it start, but the display does not scale properly for Retina screens, so the game uses only the bottom left corner. More bugs to fix...
